### PR TITLE
3248 image

### LINKED ISSUE #3166: Canvas drawing is cut off when the shape is explicitly set to be a square
- [x] I have searched to see if a similar issue already exists.

**Is your feature request related to a problem? Please describe.**  

Right now if we want to make a 512\*512 drawing canvas, the only method is (1) ask users to create a pure white 512\*512 image using third-party software like PhotoShop, then (2) ask users to import that blank image into gr.Image(source='upload', tool='sketch'), then (3) use the resulting mask as the user drawing for any applications. (Besides, the initial width of scribble can not be set by code.)

This is over-complicated. We should have an one-line function to make a simple drawing canvas.

**Describe the solution you'd like**  

We may consider something like

gr.Image(source='blank', tool='sketch', size=(512, 512))

### LINKED ISSUE #3248: UI crashes in `gr.Image` with `tool="color-sketch"` and a pre-loaded image
### Describe the bug

Running the following code
```py
#Create a black image
img = Image.new("RGB", (512, 512), (0, 0, 0))
img.save("image.png", "PNG")
#Pre-load the black image in the component
gr.Image(value="image.png", interactive=True, tool="color-sketch")
```
Crashes the UI
image

### Reproduction

Here's a colab with an example: https://colab.research.google.com/drive/1Gp5pp51P14B0tM331dS7476-Oqe-Bi04

### System Info

```shell
3.19.0
```

### Severity

serious, but I can work around it

### PR DESCRIPTION
**Pull request by pngwn**
# Description

There was a bug which caused the Sketch to get stuck in a resize loop causing #3248.

Also added support for the shape kwarg on the frontend with sketches. I did look at image uploads too but that requires pretty significant changes, so will leave it for now. 

I have added demos to `blocks_mask`. `2c` and `5d`. They should be testable in the PR preview.

Closes #3248. Closes #3166.

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have added a short summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md, unless the pull request is labeled with the "no-changelog-update" label.

Please add a brief summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your github profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by abidlabs** [comment]
This looks great! Can confirm demo 5d works as expected. For 2c, I think it would be helpful as a visual aid to users to know where the sketchpad starts and stops. Like maybe the component is grayed out outside of the boundaries of the sketchpad? 

Here's how it currently looks:

image

Here's what I'm proposing (but without hiding the icons):

image

**Comment by pngwn** [comment]
Yeah I agree, would a border work? I'm not a huge fan of the grey, looks very sad.

**Comment by freddyaboulton** [comment]
I like the border approach @pngwn !

**Comment by pngwn** [comment]
Did the `brush_radius` thing for @apolinario too. Can now set a default size and update it via `gr.update()`

Can test like this when i push:

```py
with gr.Blocks():
  gr.Image(source="canvas", tool="color-sketch")
  gr.Image(source="canvas", brush_radius=100, tool="color-sketch")
  img = gr.Image(source="canvas", interactive=True, tool="color-sketch")
  r = gr.Number(20, label="Change brush size", max="100")
  btn = gr.Button("Change Brush")

  def brushy(size):
    print(size)
    return img.update(brush_radius=size)

  btn.click(brushy, inputs=r, outputs=img)
```

**Comment by abidlabs** [comment]
Border is better!

**Comment by pngwn** [comment]
Brush size thing introduced some bugs will see if i can fix quickly.

**Comment by pngwn** [comment]
Okay i think we're good. Updated changelog again and fixed bug and added border.

This was the brush change btw:

https://user-images.githubusercontent.com/12937446/220747841-04428244-5294-4769-b8fb-b557a7d34d38.mov

**Comment by pngwn** [comment]
border:

Screenshot 2023-02-22 at 20 17 21

**Comment by pngwn** [comment]
Will fix test

File: gradio/components.py [hunk 1]
```diff
@@ -1398,7 +1399,9 @@ def __init__(
             streaming: If True when used in a `live` interface, will automatically stream webcam feed. Only valid is source is 'webcam'.
             elem_id: An optional string that is assigned as the id of this component in the HTML DOM. Can be used for targeting CSS styles.
             mirror_webcam: If True webcam will be mirrored. Default is True.
+            brush_radius: Size of the brush for Sketch. Default is None which chooses a sensible default
```
Comment (abidlabs) [inline gradio/components.py:1402]:
```suggestion
            brush_radius: Size of the brush for sketching (in pixels). Default is None which chooses a sensible default. Only applicable if `tool` is `sketch` or `color-sketch`
```

Comment (abidlabs) [inline gradio/components.py:1402]:
```suggestion
            brush_radius: Size of the brush for sketching (in pixels). Default is None which chooses a sensible default. Only applicable if `tool` is `sketch` or `color-sketch`
```

**Review by abidlabs — APPROVED** [review]
Tested and LGTM. Very cool PR @pngwn!

**Comment by Charmandrigo** [comment]
Hey, not sure what's going on but my UI is on Gradio 3.31.0 and this issue still exist, at least for me, same description as other issues, it freezes the addon while loading then it crashes. It happens when I create a new 512,512 canvas with the Two Shot addon

**Comment by pngwn** [comment]
Could you update to the latest version to see if the issues is still present?

If it is could you please create a new issue with a minimal reproduction.

**Pull request merged**